src/tcgwars/logic/impl/gen3/PopSeries2.groovy

```diff
@@ -370,7 +436,12 @@ public enum PopSeries2 implements LogicCardInfo {
         weakness P
         pokePower "Time Reversal", {
           text "Once during your turn, when you put Celebi ex from your hand onto your Bench, you may search your discard pile for a card, show it to your opponent, and put it on top of your deck."
-          actionA {
+          onActivate {r->
+            if (r==PLAY_FROM_HAND && bg.em().retrieveObject("Time_Reversal")!=bg.turnCount && my.discard.filterByType(TRAINER) && confirm('Use Time Reversal to search your discard pile for a card to put on top of your deck?')) {
```

Comment:
I don't think it's proper to use a storedObject here. Once a card goes to your hand or deck it is private information and no longer has a record of used effects. This should only be done for "You can only use 1 $name per turn" abilities.